Add applicant decisions to Portal along with logistics email

njhuey opened the pull request:
- Redesigned Portal Page into Verticle Timeline format
- Added backend implementation for Verticle Timeline variables 
- constrained allowed status values

RandyHuynh5815 commented on src/views/portal/components/VerticleTimeline.tsx:
```diff
@@ -0,0 +1,64 @@
+function VerticleTimeline(props: TimelineProps) {
+	const first_item = (
+		<ListGroup.Item className={styles.list_item}>
+			<Form.Check disabled={true} checked={true} className={styles.check} />
```
Using checkboxes may be misleading to the user since the box is checked no matter the status (as long as the stage is completed). This problem is apparent when the checkmark is the same for both accepted and denied in `second_item`. Try using the assets/icons from #72 to represent something being completed or in progress.

RandyHuynh5815 commented on src/views/portal/components/VerticleTimeline.tsx:
```diff
@@ -0,0 +1,64 @@
+	);
+
+	const second_item =
+		props.status === "submitted" ? (
```
I'm not sure nested ternary operations are good practice, though looking online, the issue seems split. @taesungh How do you feel about this?

RandyHuynh5815 reviewed (dismissed):
Thank you for getting started on this. There are some issues I would like you to take a look at before Taesung reviews this and have this branch merged.

njhuey committed:
Refactored portal page

njhuey committed:
bugfix

taesungh commented on src/views/portal/VerticalTimeline.tsx:
```diff
@@ -0,0 +1,62 @@
+					height={25}
+					className={styles.icon}
+				/>
+				Application accepted: {props.verdict_time}
```
Thought: I'm not sure how meaningful it is to actually display this verdict time.

taesungh commented on src/api/routers/user.py:
```diff
@@ -22,6 +22,8 @@
 class IdentityResponse(BaseModel):
     uid: Optional[str]
     status: Optional[str]
+    submission_time: Optional[datetime]
+    verdict_time: Optional[datetime]
```
Thought: when initially planning out the API integration, I considered having a separate endpoint for applicant-specific data rather than cluttering the global user identity. This would be helpful/cleaner if we had other user roles interacting with our site, but when discussing with directors earlier this week, most likely, only applicants and organizers (no mentors, volunteers, workshop hosts, sponsors, or judges) will need site access.

taesungh commented on src/views/portal/Message.tsx:
```diff
@@ -0,0 +1,34 @@
+import styles from "views/portal/Portal.module.scss";
+
+interface MessageProps {
+	status: string;
+}
```
Note: we enumerated some of the applicant statuses while working on the admin portal as part of #72, but I'm also reorganizing more of it while working on #84. As of now, we'll be separating status from the decision, more to come in another pull request soon.

taesungh commented on src/views/portal/Portal.tsx:
```diff
@@ -1,16 +1,16 @@
 import { useContext, useEffect } from "react";
 
 import { TitleBanner } from "components";
+import { Container } from "react-bootstrap";
```
Chore: When [importing components with React Bootstrap](https://react-bootstrap.github.io/getting-started/introduction#importing-components), we should import individual components. Also, we should try and keep the import ordering consistent, i.e. third-party in a group before first-party.

taesungh commented on src/views/portal/VerticalTimeline.tsx:
```diff
@@ -0,0 +1,62 @@
+import ListGroup from "react-bootstrap/ListGroup";
+import styles from "views/portal/Portal.module.scss";
+
+export interface TimelineProps {
```
Question: is this interface used anywhere else so that it needs to be exported?
Nitpick: To follow convention throughout our components, we should name the props interface based on the component name, i.e. `VerticalTimelineProps`.

taesungh commented on src/views/portal/VerticalTimeline.tsx:
```diff
@@ -0,0 +1,62 @@
+	return (
+		<div className={styles.verticle_timeline}>
+			<ListGroup className={styles.list}>
+				{first_item}
+				{second_item}
```
Suggestion: we could probably use more descriptive names for these items.

taesungh commented on src/views/portal/VerticalTimeline.tsx:
```diff
@@ -0,0 +1,62 @@
+	verdict_time?: string | null;
+}
+
+function VerticalTimeline(props: TimelineProps) {
```
Suggestion: we can destructure individual properties rather than referencing the entire object throughout the component.

taesungh commented on src/views/portal/VerticalTimeline.tsx:
```diff
@@ -0,0 +1,62 @@
+		</ListGroup.Item>
+	);
+
+	const second_item =
+		props.status === "accepted" || props.status === "confirmed" ? (
```
Note: based on current reviews from the applications committee, we will also need to handle when an applicant is waitlisted.

taesungh replied:
Update: actually, we won't have a waitlist decision, just accepted or rejected.

taesungh reviewed (changes requested):
Thanks for continuing to work on this. I'm working on reorganizing how we handle applicant status and decision while redesigning the admin interface. Logistics should provide the actual message content for each decision type soon. In the mean time, there are some things we could clean up.

njhuey committed:
temp commit

njhuey committed:
Merge branch 'main' of github.com:HackAtUCI/HackUCI-Site into feature/portal

njhuey committed:
added acceptance email

njhuey committed:
bugfix

njhuey committed:
bugfix pt2

taesungh commented on src/views/portal/Message.tsx:
```diff
@@ -0,0 +1,173 @@
+					look forward to seeing you at Hack at UCI!
+				</p>
+
+				<p>
+					<h4>Waiver</h4>
```
Chore: invalid DOM nesting
> `<h4>` cannot appear as a descendant of `<p>`

njhuey committed:
fixed hydration error

taesungh commented on src/views/portal/Message.tsx:
```diff
@@ -0,0 +1,169 @@
+					Parking Structure. This structure provides both pay-by-space ($2 per
+					hour) and permit dispensers ($1 per hour, $13 for a day). You may
+					purchase a permit at any information kiosk, dispenser, or through
+					UCI&#39;s ParkbyPlate signage located in select parking areas. All
```
Nitpick [repeating]: although `&#39;` works here, `&apos;` is probably more meaningful to a developer.

taesungh commented on src/views/portal/Message.tsx:
```diff
@@ -0,0 +1,169 @@
+					Map:{" "}
+					<a href="https://map.concept3d.com/?id=463#!ce/8403?ct/55161,27429,35849?m/367524?s/">
+						https://map.concept3d.com/?id=463#!ce/8403?ct/55161,27429,35849?m/367524?s/
+					</a>
```
Similar, could use more descriptive link text.

njhuey committed:
added confirmed message

taesungh commented on src/views/portal/Message.tsx:
```diff
@@ -0,0 +1,169 @@
+					<strong>Attendees will be responsible for their own parking.</strong>{" "}
+					The closest parking structure to our event is the UCI Student Center
+					Parking Structure. This structure provides both pay-by-space ($2 per
+					hour) and permit dispensers ($1 per hour, $13 for a day). You may
```
Nitpick: inconsistent time units, i.e. use "per" or "for" in both

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+				Still looking for teammates? Attend our two team formation events,
+				Thursday (2/2) 7-8 PM in DBH 6011 and Friday (2/3) 7-8 PM in Student
+				Center Pacific Ballroom. You can also create an introductory slide{" "}
+				<a href="https://docs.google.com/presentation/d/1qjAzK9ocrI5Lg2D8l4VnE38CWDU2bW2v3XqAROA-vR4/edit#slide=id.g1ecf0247681_1_277">
+					here
+				</a>{" "}
```
Suggestion: instead of writing "here", we could use more descriptive link text, e.g.
> create an introductory slide [on our team formation slides](https://github.com/HackAtUCI/HackUCI-Site/pull/78/files#)

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+				<strong>Attendees will be responsible for their own parking.</strong>{" "}
+				The closest parking structure to our event is the UCI Student Center
+				Parking Structure. This structure provides both pay-by-space ($2 per
+				hour) and permit dispensers ($1 per hour, $13 for a day). You may
```
Nitpick: inconsistent time units, i.e. use "per" or "for" in both

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+				Parking Structure. This structure provides both pay-by-space ($2 per
+				hour) and permit dispensers ($1 per hour, $13 for a day). You may
+				purchase a permit at any information kiosk, dispenser, or through
+				UCI&#39;s ParkbyPlate signage located in select parking areas. All
```
Nitpick [repeating]: although &#39; works here, &apos; is probably more meaningful to a developer.

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+				via Slack, and we&#39;d be happy to help.
+			</p>
+			<p>
+				Map:{" "}
+				<a href="https://map.concept3d.com/?id=463#!ce/8403?ct/55161,27429,35849?m/367524?s/">
+					https://map.concept3d.com/?id=463#!ce/8403?ct/55161,27429,35849?m/367524?s/
+				</a>
```
Similar, could use more descriptive link text.

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+				<li>Sign the liability waiver once it has been sent to your email</li>
+				<li>Join the Slack once the invite has been sent to your email</li>
+				<li>
+					Bring a <strong>photo ID for check</strong> in on Day 1, and your{" "}
```
Question: should "check-in" also be rendered as important? Also, since check-in is used as a noun here, it should be hyphenated.

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+			<h4>Questions/Comments</h4>
+			<p>
+				If you have any questions that have not yet been answered, please reach
+				out to us at <u>hack@uci.edu</u>, and we will be glad to assist you with
```
Chore: we should add a proper link to our email using `mailto:`.

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+				</a>{" "}
+				and sign the Event Waiver prior to the event. You will receive the
+				waiver via email closer to the event date.{" "}
+				<strong className={styles.red_text}>
+					<i>
+						<u>
+							You must complete the waiver in order to participate in Hack at
+							UCI.
+						</u>
+					</i>
+				</strong>
```
Thought: The underline might be a bit excessive (I think a good practice is reserving the use of underline for hyperlinks)

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+			<h4>Team Formation</h4>
+			<p>
+				Still looking for teammates? Attend our two team formation events,
+				Thursday (2/2) 7-8 PM in DBH 6011 and Friday (2/3) 7-8 PM in Student
```
Suggestion: there are a lot of numbers here, maybe we could write out or abbreviate the month name instead of using the number?

taesungh reviewed (changes requested):
Few things we should address about the actual acceptance message text. @rileysw, could you handle some of these?

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,139 @@
+				. Hackathon events will take place in the following rooms:
+			</p>
+			<p>
+				<strong>Pacific Ballroom Lobby:</strong> Check In
+				<br />
+				<strong>Pacific Ballroom ABCD:</strong> Hacking Space
+				<br />
+				<strong>Woods Cove B + C:</strong> Hacking Space
+				<br />
+				<strong>Moss Cove A + B:</strong> Workshops and Socials
```
Suggestion: these locations could probably be displayed as an unordered list.

taesungh commented on src/views/portal/Message.tsx:
```diff
@@ -0,0 +1,51 @@
+					your attendance, it is crucial that you read through the following
+					event logistics in its entirety and complete the required tasks. We
+					look forward to seeing you at Hack at UCI!
+				</p>
+				<AcceptedMessage />
```
Chore: The previous heading is `<h2>Status</h2>` while the accepted message includes `h4`. We should include an `h3` element in between, perhaps saying "Event Logistics".

njhuey committed:
added schedule to accepted message

rileysw committed:
updated portal message content

rileysw commented on src/views/portal/Message.tsx:
```diff
@@ -0,0 +1,169 @@
+					<li>Sign the liability waiver once it has been sent to your email</li>
+					<li>Join the Slack once the invite has been sent to your email</li>
+					<li>
+						Bring a <strong>photo ID for check</strong> in on Day 1, and your{" "}
```
I bolded only photo ID so it's consistent with the bold on the badge

rileysw committed:
removed sat and sun food from portal schedule

njhuey committed:
implemented rsvp endpoint

njhuey committed:
Merge branch 'feature/portal' of github.com:HackAtUCI/HackUCI-Site into feature/portal

njhuey committed:
added rsvp endpoint

taesungh commented on src/api/models/ApplicationData.py:
```diff
@@ -41,5 +41,4 @@ class Config:
 
 class ProcessedApplicationData(RawApplicationData):
     resume_url: Optional[HttpUrl]
-    submission_time: datetime
```
We still need to keep this attribute to process a submitted application.

taesungh commented on src/api/routers/user.py:
```diff
@@ -123,3 +123,27 @@ async def apply(
+    if not user_record:
+        raise HTTPException(status.HTTP_400_BAD_REQUEST)
+
+    if user_record["status"] == "accepted":
+        new_status = "confirmed"
+    elif user_record["status"] == "confirmed":
+        new_status = "accepted"
```
Chore: We should use the proper enumerated values for status/decision.

rileysw committed:
fixed team formation portal wording

taesungh committed:
Restore `submission_time` in application data

taesungh committed:
Use Status and Decision Enums in RSVP endpoint

taesungh committed:
Enumerate Portal status on frontend

taesungh committed:
Fix RSVP form and endpoint

taesungh commented on src/views/portal/AcceptedMessage.tsx:
```diff
@@ -0,0 +1,167 @@
+
+			<h4>Pre-Hack-At-UCI Checklist</h4>
+			<ol>
+				<li>Confirm your attendance by clicking on the button below</li>
```
Need to fix wording here

rileysw replied:
I updated it to `Confirm your attendance by clicking on the RSVP button at the bottom
					of the page`. I'm not quite sure if this is what you had in mind.

taesungh replied:
Yeah, that's fine for now.

rileysw committed:
fixed wording for hacker checklist

taesungh committed:
Graduate heading weights in museum container

taesungh committed:
Include other small adjustments to Portal

- Fix miscellaneous grammar/style issues on acceptance message
- Include `<hr>` before RSVP
- Use Bootstrap's `text-danger` for red text

taesungh commented:
Prematurely deployed as live for decision release.
